lib/puppet/type/bmc_network.rb

```diff
@@ -20,36 +20,46 @@
   newproperty(:ipaddr) do
     desc 'Ip Address'
     validate do |value|
-      raise ArgumentError, "%s is not a valid ip address" % value unless (value =~ Resolv::IPv4::Regex || value =~ Resolv::IPv6::Regex)
+      unless (value =~ Resolv::IPv4::Regex || value =~ Resolv::IPv6::Regex)
+        raise Puppet::Error, "%s is not a valid ip address" % value
+      end
```

Comment:
Create a define and call it. You are using the same unless statement to raise and error four times.